Sprint_2

natasha1202 opened the pull request:
BooksCollector tests added

natasha1202 committed:
BooksCollector tests added

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,23 @@
+@pytest.fixture
+def book_collector():
+    book_collector = BooksCollector()
+    book_collector.books_genre = {'Дом огней': 'Детективы',
```
Нужно исправить: здесь и далее по тестам, все прямые обращения к атрибутам класса необходимо переписать на методы, для вызова этих атрибутов.
 Использовать прямое обращение только в крайних случаях, когда метод для этого атрибута не написан.
 Мы тестируем программу, нам важно что не просто атрибут изменился, а метод для работы с этим атрибутом тоже работает

natasha1202 committed:
Readme updated
corrected test_get_list_of_favorites_books_existing_favorite_list, test_add_book_in_favorites_book_already_in_favorites

natasha1202 committed:
attributes replaced with methods

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,71 @@
+        collector = BooksCollector()
+        collector.add_new_book('Происхождение')
+        collector.set_book_genre('Происхождение', 'Детективы')
+        assert collector.books_genre.get('Происхождение') == 'Детективы'
```
get_book_genre()

natasha1202 committed:
test_set_book_genre_to_one_existing_book corrected, attribute replaced with getter
